cinema/models.py

```diff
@@ -1,3 +1,53 @@
 from django.db import models
+from django.utils.functional import cached_property
 
 # Create your models here.
+
+
+class Movie(models.Model):
+    name = models.CharField(max_length=50)
+    poster = models.ImageField(upload_to='images')
+    description = models.CharField(max_length=1000)
+    imdb_id = models.CharField(max_length=30)
+    trailer_link = models.CharField(max_length=150)
+    length = models.CharField(max_length=10)
+
+    def __str__(self):
+        return self.name
+
+    @cached_property
+    def imdb_link(self):
+        return 'https://www.imdb.com/title/' + self.imdb_id.__str__()
+
+
+class Cinema(models.Model):
+    name = models.CharField(max_length=30)
+    description = models.CharField(max_length=300)
+    city = models.CharField(max_length=30)
+    address = models.CharField(max_length=70)
+    hall = models.CharField(max_length=30)
+
+    def __str__(self):
+        return self.name
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=30)
+    seats = models.IntegerField()
+    description = models.CharField(max_length=300)
+
+    def __str__(self):
+        return self.name
+
+
+class Seat(models.Model):
+
+    OCCUPIED = 'O'
+    EMPTY = 'E'
+    STATUS_CHOICES = [
+        (OCCUPIED, 'Occupied'),
+        (EMPTY, 'Empty')
+    ]
+
+    reservation = models.CharField(max_length=10, choices=STATUS_CHOICES, default=EMPTY)
```

Comment:
Here, reservation is going to be a new table, so it doesn't make much sense to add it as a `CharField`. A seat is part of a hall also. Check the relationship between the tables and remake the models accordingly.